src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs

```diff
@@ -70,7 +71,7 @@ public static class WebScriptHostBuilderExtension
                     }
 
                     // Logging and diagnostics
-                    services.AddSingleton<IMetricsLogger, WebHostMetricsLogger>();
+                    services.TryAddSingleton<IMetricsLogger, WebHostMetricsLogger>();
```

Comment:
Wanted to call out this change -- In order to override IMetricsLogger in e2e tests, I had to change this.